blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -48,13 +50,18 @@ object BlazeClient {
               .flatMap { _ => manager.release(connection) }
             Task.now(DisposableResponse(r, dispose))
 
-          case -\/(Command.EOF) =>
+          case -\/(Command.EOF) if !isFresh =>
             invalidate(connection).flatMap { _ =>
-              manager.borrow(key).flatMap { newConn =>
-                loop(newConn, flushPrelude)
+              manager.borrow(key, true).flatMap { newConn =>
+                loop(newConn, flushPrelude, true)
```

Comment:
This is going to be less aggressive in evicting stale connections.  We might now open a connection when there is a perfectly fine one next in the queue.  What I have not done is measure how long it takes to loop through several stale connections, so this may or may not be a good thing.

With @alissapajer's idea on borrow, we could loop the old way, and still have the termination condition we need.  Instead of asking for a fresh one, we'd be told whether it's a fresh one.